src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryToActivityConverter.cs

```diff
@@ -20,16 +20,21 @@
 using System.Reflection.Emit;
 using OpenTelemetry.Trace;
 using StackExchange.Redis.Profiling;
+#if NET6_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+#endif
 
 namespace OpenTelemetry.Instrumentation.StackExchangeRedis.Implementation;
 
 internal static class RedisProfilerEntryToActivityConverter
 {
+    private const string ScriptEvalMessageTypeName = "StackExchange.Redis.RedisDatabase+ScriptEvalMessage";
+
     private static readonly Lazy<Func<object, (string?, string?)>> MessageDataGetter = new(() =>
     {
-        var redisAssembly = typeof(IProfiledCommand).Assembly;
-        Type profiledCommandType = redisAssembly.GetType("StackExchange.Redis.Profiling.ProfiledCommand");
-        Type scriptMessageType = redisAssembly.GetType("StackExchange.Redis.RedisDatabase+ScriptEvalMessage");
+        Type profiledCommandType = Type.GetType("StackExchange.Redis.Profiling.ProfiledCommand, StackExchange.Redis", throwOnError: true)!;
+        Type scriptMessageType = Type.GetType(ScriptEvalMessageTypeName + ", StackExchange.Redis", throwOnError: true)!;
```

Comment:
Remove ` + `?

Reply:
The AOT/trimming tools don't recognize patterns like 

```C#
var redisAssembly = typeof(IProfiledCommand).Assembly;
redisAssembly.GetType("StackExchange.Redis.Profiling.ProfiledCommand");
```

The don't know which Type you are referring to when you do it this way. But if you instead say:

```C#
Type.GetType("StackExchange.Redis.Profiling.ProfiledCommand, StackExchange.Redis");
```

They do recognize this pattern and know exactly which type you are talking about.

The `+` is added above because I refactored out the `StackExchange.Redis.RedisDatabase+ScriptEvalMessage` name into a constant string so I could use it in 2 places:

1. In the Type.GetType call
2. In the `[DynamicDependency("CommandAndKey", ScriptEvalMessageTypeName, "StackExchange.Redis")]` line.